### PR TITLE
Migrate regression task.

### PR DESCRIPTION
**Pull request by alhijazi**

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
Is this a draft or ready?

File: src/clusterfuzz/_internal/protos/uworker_msg.proto [hunk 1]
```diff
@@ -194,7 +199,12 @@ enum ErrorType {
   PROGRESSION_BUILD_SETUP_ERROR = 18;
   REGRESSION_REVISION_LIST_ERROR = 19;
   REGRESSION_BUILD_NOT_FOUND = 20;
-  SYMBOLIZE_BUILD_SETUP_ERROR = 21;
+  REGRESSION_BUILD_SETUP_ERROR = 21;
+  REGRESSION_BAD_BUILD_ERROR = 22;
+  REGRESSION_NO_CRASH =23;
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/protos/uworker_msg.proto:226]: nit: missing space.

File: src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py [hunk 2]
```diff
@@ -60,18 +60,47 @@ class TestcaseReproducesInRevisionTest(unittest.TestCase):
...
+                     uworker_msg_pb2.REGRESSION_BUILD_SETUP_ERROR)
+
+  def test_bad_build_error(self):
+    """Tests _testcase_reproduces_in_revision behaviour on bad builds."""
+    self.mock.check_app_path.return_value = True
+    build_data = uworker_msg_pb2.BuildData(
+        revision=1,
+        is_bad_build=True,
+        should_ignore_crash_result=False,
+        build_run_console_output='')
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py:94]: WHy set this?

File: src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py [hunk 3]
```diff
@@ -91,45 +120,57 @@ def setUp(self):
     self.revision_list = [1, 2, 5, 8, 9, 12, 15, 19, 21, 22]
 
   def test_near_max_revision(self):
-    """Ensure that we return True if this is a very recent regression."""
+    """Ensure that we return a result if this is a very recent regression."""
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py:123]: Ensures

Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py:123]: Also change "we" to be more descriptive.

File: src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py [hunk 4]
```diff
@@ -294,47 +357,63 @@ def find_regression_range(uworker_input: uworker_msg_pb2.Input,
...
+      return error
 
     if is_crash:
       max_index = middle_index
     else:
       min_index = middle_index
 
-    _save_current_regression_range_indices(
-        testcase.key.id(), revision_list[min_index], revision_list[max_index])
+    # save current regression range in case the task dies prematurely.
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py:394]: Begin comments with capital letters.

File: src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py (hunk 2 above)
Comment (alhijazi) [reply src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py:94]: `testcase_manager.check_for_bad_build` is supposed to return a build data. I'm mocking it here to make it return a bad build and then verify that the right info is passed back through the `regression_task_output`.

Comment (alhijazi) [reply src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py:94]: I will remove `build_run_console_output=''` as it's unnecessary.

**Comment by alhijazi** [comment]
> Is this a draft or ready?

Marked as ready.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Pull request merged**